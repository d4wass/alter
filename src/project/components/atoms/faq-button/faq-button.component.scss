@use '../../../../base';

button{
    display: flex;
    justify-content: center;
    align-items: center;
    padding: 20px;
    border-radius: 50%;
    border: solid 1px base.$placeholder-gray;
    height: 5px;
    width: 5px;
    background-color: transparent;
    cursor: pointer;

    img{
        height: 25px;
    }
}