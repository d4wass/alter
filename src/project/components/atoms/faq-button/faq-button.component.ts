import { Component, EventEmitter, Output } from '@angular/core';

@Component({
  selector: 'app-faq-button',
  template: `
    <button (click)="handleOpen()" data-test="btn">
      <img src="assets/faq/minus-solid.svg" alt="" *ngIf="isOpen" />
      <img src="assets/faq/plus-solid.svg" alt="" *ngIf="!isOpen" />
    </button>
  `,
  styleUrls: ['./faq-button.component.scss']
})
export class FaqButtonComponent {
  isOpen: boolean = false;
  @Output() isOpenEvent: EventEmitter<boolean> = new EventEmitter<boolean>();

  handleOpen() {
    this.isOpenEvent.emit(!this.isOpen);
    this.isOpen = !this.isOpen;
  }
}
