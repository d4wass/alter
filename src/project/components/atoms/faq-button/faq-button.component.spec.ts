import { ComponentFixture, TestBed } from '@angular/core/testing';

import { FaqButtonComponent } from './faq-button.component';

describe('FaqButtonComponent', () => {
  let component: FaqButtonComponent;
  let fixture: ComponentFixture<FaqButtonComponent>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [FaqButtonComponent]
    }).compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(FaqButtonComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });
});
