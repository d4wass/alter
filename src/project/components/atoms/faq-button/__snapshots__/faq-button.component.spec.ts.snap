// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FaqButtonComponent should match to snapshot 1`] = `
FaqButtonComponent {
  "__ngContext__": 0,
  "isOpen": false,
  "isOpenEvent": EventEmitter_ {
    "__isAsync": false,
    "closed": false,
    "currentObservers": null,
    "hasError": false,
    "isStopped": false,
    "observers": [],
    "thrownError": null,
  },
}
`;
